src/main/dulwich/extract.py

```diff
@@ -0,0 +1,113 @@
+from typing import List, Dict, Tuple
+from dulwich.diff_tree import TreeChange
+from dulwich.objects import Blob
+
+import json
+import os
+
+from dulwich.repo import Repo
+from dulwich.porcelain import clone
+
+# from dulwich.patch import unified_diff
+from difflib import unified_diff
+
+TEMPLATE_PATH = "PositLib.git/"
+
+
+def count_diff(blob_old: Blob, blob_new: Blob) -> Tuple[int, int]:
+    """
+    Function compares two blobs and returns the Tuple of changes:
+    added and removed lines count.
+
+    :param blob_old: Old version blob
+    :param blob_new: New version blob
+    """
+    diffs = list(unified_diff(blob_old.data.decode().splitlines(), blob_new.data.decode().splitlines()))
+    add, delete = 0, 0
+    for diff in diffs:
+        if diff[0] == '+':
+            add += 1
+        elif diff[0] == '-':
+            delete += 1
+    return add, delete
+
+
+def process_change_diff(change: TreeChange, commit_dict: Dict, repo: Repo) -> Dict:
+    """
+    Function processes given change and fills the dictionary.
+
+    :param repo: Current repo instance
+    :param change: change to process.
+    :param commit_dict: Dictionary with information about commit.
+    """
+    # Added, deleted, modified
+    try:
+        if change.old.sha is None:
+            commit_dict['add'] = len(repo[change.new.sha].data.decode())
+            commit_dict['blob_id'] = change.new.sha.decode()
+            commit_dict['path'] = TEMPLATE_PATH + change.new.path.decode()
```

Comment:
you need to add information about repository at some stage but not `TEMPLATE_PATH` - it will not work in case of many repositories